src/pages/Trade/components/OrderBookCard/MyTrades.tsx

```diff
@@ -1,49 +1,106 @@
-	getMyTrades,
-	getIsLoadingMyTrades,
-	getIsLoadedMyTrades,
-} from 'ducks/trades/myTrades';
+import snxData from 'synthetix-data';
+import { SYNTHS_MAP } from 'constants/currency';
 
 import { RootState } from 'ducks/types';
-
-import useInterval from 'shared/hooks/useInterval';
+import { getIsWalletConnected, getCurrentWalletAddress } from 'ducks/wallet/walletDetails';
 
 import TradeHistory from './TradeHistory';
 
 import { REFRESH_INTERVAL } from './constants';
+import { HistoricalTrades } from 'ducks/trades/types';
+import QUERY_KEYS from 'constants/queryKeys';
+import { CurrencyKey } from 'constants/currency';
+
+const MAX_TRADES = 100;
 
 const mapStateToProps = (state: RootState) => ({
-	trades: getMyTrades(state),
-	isLoading: getIsLoadingMyTrades(state),
-	isLoaded: getIsLoadedMyTrades(state),
+	walletAddress: getCurrentWalletAddress(state),
+	isWalletConnected: getIsWalletConnected(state),
 });
 
-const mapDispatchToProps = {
-	fetchMyTradesRequest,
-};
-
-const connector = connect(mapStateToProps, mapDispatchToProps);
+const connector = connect(mapStateToProps);
 
 type PropsFromRedux = ConnectedProps<typeof connector>;
 
 type MyTradesProps = PropsFromRedux;
 
-const MyTrades: FC<MyTradesProps> = ({ fetchMyTradesRequest, trades, isLoading, isLoaded }) => {
-	useEffect(() => {
-		fetchMyTradesRequest();
-		// eslint-disable-next-line react-hooks/exhaustive-deps
-	}, []);
+type SettledTrade = {
+	exchangeTimestamp: number;
+	dest: CurrencyKey;
+	src: CurrencyKey;
+	amount: number;
+	rebate: number;
+	reclaim: number;
+};
+type SettledTrades = SettledTrade[];
+
+const MyTrades: FC<MyTradesProps> = ({ walletAddress, isWalletConnected }) => {
+	const settledTradesQuery = useQuery<SettledTrades, any>(
+		QUERY_KEYS.HistoricalTrades.MyTrades(walletAddress || ''),
+		() =>
+			snxData.exchanger.exchangeEntriesSettled({
+				from: walletAddress,
+				max: MAX_TRADES,
+			}),
+		{
+			refetchInterval: REFRESH_INTERVAL,
+		}
+	);
+	const tradesQuery = useQuery<HistoricalTrades, any>(
+		QUERY_KEYS.HistoricalTrades.MyTrades(walletAddress || ''),
+		() => {
+			const trades = snxData.exchanges.since({
+				fromAddress: walletAddress,
+				maxBlock: Number.MAX_SAFE_INTEGER,
+				max: MAX_TRADES,
+			});
+			for (let i = 0; i < trades.length; i++) {
+				if (!!settledTradesQuery.data) {
+					const settledTrade = settledTradesQuery.data.find(
+						(settledTrade) =>
+							trades[i].timestamp === settledTrade.exchangeTimestamp &&
+							settledTrade.dest === trades[i].toCurrencyKey &&
+							settledTrade.src === trades[i].fromCurrencyKey &&
+							trades[i].fromAmount === settledTrade.amount
+					);
 
-	useInterval(() => {
-		fetchMyTradesRequest();
-	}, REFRESH_INTERVAL);
+					trades[i].isSettled = false;
+					if (!!settledTrade) {
+						trades[i].rebate = settledTrade.rebate;
+						trades[i].reclaim = settledTrade.reclaim;
 
+						// special case for when the currency is priced in sUSD
+						const feeReclaimRebateAmount =
```

Comment:
I think there is a bug with my fee reclaim logic.  I see different data for my personal wallet on production vs my local.  Don't have a lot of context on the business logic here, does it look correct to you in the UI?

Reply:
Hmm looks like my settledTrades are coming back empty locally

Reply:
Ah missing `normalizeTrades()` I think

Reply:
The response I get doesn't match the type.  Still investigating why:
```
block: 10610633
date: Thu Aug 06 2020 21:43:32 GMT-0700 (Pacific Daylight Time) {}
feesInUSD: 1.2041677650435505
fromAddress: "0x2db6f5e838ed2bad993e9ff2d3d7a5c1cc35704c"
fromAmount: 401.38925501451644
fromAmountInUSD: 401.38925501451644
fromCurrencyKey: "sUSD"
fromCurrencyKeyBytes: "0x7355534400000000000000000000000000000000000000000000000000000000"
gasPrice: 28
hash: "0x65ff5072946ef6d1d755c0f5b3555c05c110911b937515d574c08c9c18274e94"
timestamp: 1596775412000
toAddress: "0x2db6f5e838ed2bad993e9ff2d3d7a5c1cc35704c"
toAmount: 0.1285246326950302
toAmountInUSD: 400.18508724947293
toCurrencyKey: "sDEFI"
toCurrencyKeyBytes: "0x7344454649000000000000000000000000000000000000000000000000000000"
```

Reply:
missing `rebate` etc, need to trace where this is being populated.

Reply:
ah Im not waiting for the promise to return